internal/preflight/preflight.go

```diff
@@ -68,15 +67,7 @@ func (p *preflight) verifyImage(ctx context.Context, mapping *kmmv1beta1.KernelM
 	moduleName := mod.Spec.ModuleLoader.Container.Modprobe.ModuleName
 	baseDir := mod.Spec.ModuleLoader.Container.Modprobe.DirName
 
-	var registryAuthGetter auth.RegistryAuthGetter
-	if mod.Spec.ImageRepoSecret != nil {
-		namespacedName := types.NamespacedName{
-			Name:      mod.Spec.ImageRepoSecret.Name,
-			Namespace: mod.Namespace,
-		}
-		registryAuthGetter = p.authFactory.NewRegistryAuthGetter(p.client, namespacedName)
-	}
-
+	registryAuthGetter := p.authFactory.NewRegistryAuthGetterFrom(mod)
```

Comment:
There is a change in logic here. Is it the behavior we want?

Before:
We were creating a `RegistryAuthGetter` if there were some `ImageRepoSecret` in the mod otherwise we were accessing the registry without auth.

Now:
We are creating a `RegistryAuthGetter` if there were some `ImageRepoSecret` in the mod otherwise we are creating a `ServiceAccountRegistryAuthGetter` that is not being used.

Reply:
The previous code was cherry-picked from upstream. In downstream the logic for authentication is different than in upstream, and this PR synchronizes the preflight authentication with downstream

Reply:
How will you access a public registry? Do we want to force the user to specify `imageRepoSecret` or `serviceAccount` in the CR even if he doesn't need it?

Reply:
we don't force user. In  case imageRepoSecret is not specified, the code will use the build ServiceAccount defined for each namespace by OCP. This logic is exactly the same that is already implemented in checkImageExists

Reply:
So it that case, if the user specify a serviceAccount in the CR, we will ignore it because we are taking the default service account?

Reply:
yes. user's SA is for security reasons, not for builder

Reply:
Ack.